### PR TITLE
feat: add binpack api

### PR DESCRIPTION
**Pull request by Poor12**

### CODE REVIEW

File: pkg/apis/core/v1alpha1/types_propagationpolicy.go [hunk 1]
```diff
@@ -120,6 +123,16 @@ const (
...
 )
 
+// ReplicasStrategy determines the strategy used by the scheduler when assigning replicas.
+// +kubebuilder:validation:Enum=PreferenceBinpack
+type ReplicasStrategy string
+
+const (
+	// ReplicasStrategyPreferenceBinpack means the replicas will be divided into clusters
+	// as few as possible, while considering clusters' capacity during the division.
+	ReplicasStrategyPreferenceBinpack ReplicasStrategy = "PreferenceBinpack"
```
Comment (Poor12) [inline pkg/apis/core/v1alpha1/types_propagationpolicy.go:133]: I think preferred is better than preference.

File: pkg/apis/core/v1alpha1/types_propagationpolicy.go [hunk 2]
```diff
@@ -120,6 +123,16 @@ const (
 	SchedulingModeDivide SchedulingMode = "Divide"
 )
 
+// ReplicasStrategy determines the strategy used by the scheduler when assigning replicas.
```
Comment (gary-lgy) [inline pkg/apis/core/v1alpha1/types_propagationpolicy.go:126]:
```suggestion
// ReplicasStrategy is the strategy used by the scheduler when assigning replicas.
```

File: pkg/apis/core/v1alpha1/types_propagationpolicy.go (hunk 1 above)
Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_propagationpolicy.go:133]: +1

Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_propagationpolicy.go:133]: Actually, `binpack` would do.

Comment (Poor12) [reply pkg/apis/core/v1alpha1/types_propagationpolicy.go:133]: +1. I like the brief one.

File: pkg/apis/core/v1alpha1/types_propagationpolicy.go [hunk 3]
```diff
@@ -67,6 +67,11 @@ type PropagationPolicySpec struct {
 	// SchedulingMode determines the mode used for scheduling.
 	SchedulingMode SchedulingMode `json:"schedulingMode"`
 
+	// ReplicasStrategy is the strategy used for scheduling replicas.
+	// +optional
+	// +kubebuilder:default:=Weighted
```
Comment (gary-lgy) [inline pkg/apis/core/v1alpha1/types_propagationpolicy.go:72]:
```suggestion
	// +kubebuilder:default:=Spread
```

Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_propagationpolicy.go:72]: `Weighted` doesn't really communicate `spread`

Comment (Poor12) [reply pkg/apis/core/v1alpha1/types_propagationpolicy.go]: Updated

**Pull request merged**